src/PhpDoc/TypeNodeResolver.php

```diff
@@ -358,9 +361,32 @@ private function resolveIdentifierTypeNode(IdentifierTypeNode $typeNode, NameSco
 			return new ErrorType();
 		}
 
+		if ($this->mightBeConstant($typeNode->name)) {
+			$constType = $this->tryResolveConstant($stringName) ?? $this->tryResolveConstant($typeNode->name);
+			if ($constType !== null) {
+				return $constType;
+			}
+		}
+
 		return new ObjectType($stringName);
 	}
 
+	private function mightBeConstant(string $name): bool
+	{
+		return preg_match('(^[A-Z_][A-Z0-9_]*$)', $name) > 0;
```

Comment:
since `A` and `B` or similar might be a constant in this case, I think `tryResolveConstant` should at first try whether there is a class defined with this name. if not, try its actually a constant.

what do you think?

Reply:
Agreed, constant can be tried only when it's UPPER_CASE and class of that name does not exist.